src/vec_map.rs

```diff
@@ -0,0 +1,50 @@
+use serde::{ser::SerializeMap, Serialize, Serializer};
+use std::collections::HashMap;
+
+/// Use a sorted Vec to store key-value pairs.
+/// Useful as a replacement of HashMap/BTreeMap where:
+/// - There is no need for fast access of values by keys.
+/// - The keys need to be sorted to be use as partition key
+/// for PartitionBatchSink.
+///
+/// Ideal to store labels map in various sinks.
+#[derive(Debug, Clone)]
+pub struct VecMap<V> {
+    pub entries: Vec<(String, V)>,
```

Comment:
If this has an invariant of staying sorted, it should probably not be `pub` to avoid other code from breaking that invariant.